packages/@atjson/document/src/annotation.ts

```diff
@@ -4,6 +4,24 @@ import Change, { AdjacentBoundaryBehaviour, Deletion, Insertion } from './change
 import Document from './index';
 import JSON from './json';
 
+function areAttributesEqual(lhsAnnotationAttributes: any, rhsAnnotationAttributes: any): boolean {
+  for (let key in lhsAnnotationAttributes) {
+    if (lhsAnnotationAttributes[key] !== rhsAnnotationAttributes[key]) {
+      if (lhsAnnotationAttributes[key] instanceof Document && rhsAnnotationAttributes[key] instanceof Document) {
```

Comment:
You're accessing the values of these objects multiple times; it may be best to access it once, then compare.